Docs - add rocPyJPEG (#193)
ROCm/rocPyDecode: docs/reference/rocPyJPEGDecode-api.rst

.. meta::
  :description: rocPyJPEGDecode API documentation
  :keywords: rocPyJPEGDecode, rocDecode, rocJPEG, ROCm, API, documentation, video, decode, decoding, acceleration

*********************************
rocPyJPEGDecode Python API
*********************************

> The following rocPyJPEGDecode API calls are used to decode a JPEG image. 

Reviewer: ```suggestion
Use the following rocPyJPEGDecode API calls to decode a JPEG image. 

```

Reviewer: Do we ever have to describe what constitutes a JPEG file?

Author: I don't think so but I'll look into it. This still needs to be reviewed by the SME. 

I left the wording as-is though because I'm not entirely sure that there aren't other ways to decode the image.

``initialize_hip()``
	Initializes the HIP device for decoding. The user can specify the devide to use for decoding. By default device 0 is used. 
	
	``initialize_hip()`` must be called before any other call is made to the rocPyJpeg library.
		
``decoder()``
	Constructor for the main decoding class.

``decoder().set_output_image_format()``
	Passes the output image format to the decoder engine. The output image format can be set to ``ROCJPEG_OUTPUT_RGB`` or ``ROCJPEG_OUTPUT_RGB_PLANAR``. These types are defined in ``pyRocJpegDecode/types.py``. 
		
``decoder().read`` and 	``decoder().decode``

> 	Both these functions read in a JPEG image or batch of JPEG images to be decoded and pass them to the decoder, then return the decoded images. 

Reviewer: ```suggestion
	Both functions read in a JPEG image or a batch of JPEG images to be decoded and pass them to the decoder, then return the decoded images. 

```

Author: I like my version better if it's all the same.

	
	If only one image is passed to the function, it will return one image tensor in the specified output format. If more than one image is passed to the function, it will return a list of image tensors in the specified output format.

> 	If any image fails to be decoded because it's in an unsupported format or is corrupted, no image will be returned and ``decode`` will produce an error message. 

Reviewer: ```suggestion
	If any image fails to be decoded because it's in an unsupported format or is corrupted, no image will be returned, and ``decode`` will produce an error message. 

```

Author: I disagree about this comma.

	Input can be any of the following:

	* The full path to one image.
	* The full path to a folder containing multiple images. The batch size must also be passed to the function.
	* One image stored as a memory buffer

> 	* Multiple images stored as memory buffers. The batch size must also be passed to the function.
> 	* One image stored as a NumPy array.

Reviewer: ```suggestion
	* Multiple are images stored as memory buffers. The batch size must also be passed to the function.
	* One image is stored as a NumPy array.

```

Author: The "is" changes the meaning. I'm leaving it as-is. as is. get it?

> 	* Multiple images stored as NumPy arrays. The batch size must also be passed to the function.

Reviewer: ```suggestion
	* Multiple images are stored as NumPy arrays. The batch size must also be passed to the function.

```

Author: same as before. this changes the meaning of the sentence.
